components/create-reminder-modal.tsx

```diff
@@ -0,0 +1,195 @@
+'use client';
+
+import Select from 'react-select';
+import { useEffect } from 'react';
+import { Input } from './input';
+import { CreateReminderModalProps } from '@/models/global-types';
+import { CREATE_LEAD_STATUS_NEW } from '@/utils/constants/common-constants';
+import { TextArea } from './text-area';
+import { Button } from './button';
+import './dropdown-select.css';
+import React from 'react';
+import Modal from 'react-modal';
+import Datetime from 'react-datetime';
+import 'react-datetime/css/react-datetime.css';
+import { ClockIcon } from '@/assets/icons';
+
+if (Modal.defaultStyles.overlay) {
+  Modal.defaultStyles.overlay.backgroundColor = '#00000054';
+}
+
+const CreateReminderModal = ({
+  modalIsOpen,
+  setModalIsOpen = () => {},
+  formData,
+  setFormData = () => {},
+  formErrors,
+  setFormErrors = () => {},
+  selected,
+  setSelected = () => {},
+}: CreateReminderModalProps) => {
+  const handleInputChange = (e: any) => {
+    const { name, value } = e.target;
+
+    setFormData((prev: any) => {
+      return { ...prev, [name]: value };
+    });
+
+    setFormErrors((prev: any) => {
+      return { ...prev, [name]: '' };
+    });
+  };
+
+  useEffect(() => {
```

Comment:
I hope we can move this `useEffect` at the top level of this component, i.e. we can put it above `handleInputChange` handler.